### ISSUE TITLE
Collections are not scrolling if scroll on body in the Chrome browser

### ISSUE THREAD

**Issue by ValeraS** [adobe/react-spectrum#8977]
### Provide a general summary of the issue here

After changes in #8715, keyboard navigation does not scroll collection's items into viewport if scroll on the body in Chrome browser.

### 🤔 Expected Behavior?

Collection's items scroll into viewport.

### 😯 Current Behavior

subj

### 💁 Possible Solution

https://github.com/adobe/react-spectrum/blob/76e8dd1c502ab7d5e9ce2edf413049ce2d20e95c/packages/%40react-aria/utils/src/scrollIntoView.ts#L152
```js
let scrollParents = getScrollParents(targetElement);
if (!isScrollPrevented) {
    scrollParent.push(root);
}
```

### 🖥️ Steps to Reproduce

https://codesandbox.io/p/sandbox/xenodochial-dream-kdff5w

https://github.com/user-attachments/assets/67b62f13-8db6-4823-a623-fd6c83bb49fb

### Version

react-aria-components@1.13.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

macos

**Comment by LFDanLu** [comment]
Thanks for catching this, reproduced locally. I think your proposed fix makes sense since we were previously relying on the assumption that that portion of the scroll code would only fire if scrolling was being prevented, would you like to make a PR for it?

**Comment by ValeraS** [comment]
@LFDanLu Hi, it's again does not work on react-aria-components v.1.16.0

**Comment by LFDanLu** [comment]
@ValeraS thanks for catching this, looks like it broke as a part of https://github.com/adobe/react-spectrum/pull/9146. We actually don't include the root as a part of the scrollParents anymore due to the change in https://github.com/adobe/react-spectrum/blob/4b48a5016dba403a0363a78676d94eb03f85bd1f/packages/%40react-aria/utils/src/getScrollParents.ts#L19-L24, so that will need to be updated to bring that back. 

However, doing that seems to be insufficient. It now seems to attempt to scroll the table body itself as you get partially down the table and triggers https://github.com/adobe/react-spectrum/blob/4b48a5016dba403a0363a78676d94eb03f85bd1f/packages/%40react-aria/utils/src/scrollIntoView.ts#L159-L164
even though the item you are keyboard navigating to is already in view. I think https://github.com/adobe/react-spectrum/blob/4b48a5016dba403a0363a78676d94eb03f85bd1f/packages/%40react-aria/utils/src/scrollIntoView.ts#L76 needs to return 0 if the scrollView is the root since we already use clientHeight here: https://github.com/adobe/react-spectrum/blob/4b48a5016dba403a0363a78676d94eb03f85bd1f/packages/%40react-aria/utils/src/scrollIntoView.ts#L49
which should exclude the horizontal scrollbar. That seems to improve behavior but the scrolling isn't quite working, it scrolls to the proper position but then undos its progress with another scrollIntoView call

**Comment by LFDanLu** [comment]
I've opened a PR for this https://github.com/adobe/react-spectrum/pull/9780. Still needs to go through some through testing/logic check, feel free to take a look if you'd like

**Comment by LFDanLu** [comment]
This got fixed by https://github.com/adobe/react-spectrum/pull/9784 as a side effect, but https://github.com/adobe/react-spectrum/pull/9780 has some extra fixes with the scroll into view calculations
